src/Microsoft.Data.Analysis/PrimitiveDataFrameColumn.cs

```diff
@@ -775,5 +775,30 @@ protected internal override Delegate GetDataViewGetter(DataViewRowCursor cursor)
 
         private static ValueGetter<double> CreateDecimalValueGetterDelegate(DataViewRowCursor cursor, PrimitiveDataFrameColumn<decimal> column) =>
             (ref double value) => value = (double?)column[cursor.Position] ?? double.NaN;
+
+        private ValueGetter<T> getter = null;
+
+        protected internal override void AddValueUsingCursor(DataViewRowCursor cursor, DataViewSchema.Column column)
+        {
+            long row = cursor.Position;
+            T value = default;
+            if (getter == null)
+            {
+                getter = cursor.GetGetter<T>(column);
+            }
```

Comment:
Could you have a prep step or something like that so that you don't have to check for the getter each time?

Reply:
Yea I wasn't very happy about the check for each iteration either. I have an alternative up now.